Modules/_ctypes/stgdict.c

```diff
@@ -340,6 +340,48 @@ MakeAnonFields(PyObject *type)
     return 0;
 }
 
+/*
+  Compute ceil(log10(x)), for the purpose of determining string lengths.
+*/
+static Py_ssize_t
+clog10(Py_ssize_t n)
+{
+    Py_ssize_t log_n = 0;
+    while (n > 0) {
+        log_n++;
+        n /= 10;
```

Comment:
Multiplication is often faster than division.  Can this be rewritten by computing powers of 10 until n is exceeded?

Better yet, just inline linear search.

```c
   if (n < 10ULL)
       return 1;
   if (n < 100ULL)
       return 2;
    ...
   if (n < 10_000_000_000_000_000_000ULL)
       return 20; 
```

Reply:
On the second thought, I would be surprised if this has not been implemented elsewhere in cpython code base.  Off the top of my head, I cannot recall where it could be, but I will try to search.  If someone beats me to it - please leave a note.

Reply:
> Can this be rewritten by computing powers of 10 until n is exceeded?

This is risky because the power can overflow

Reply:
> I would be surprised if this has not been implemented elsewhere in cpython code base. Off the top of my head, I cannot recall where it could be, but I will try to search.

It took me a little longer than I expected to get back to this, but the code that I was looking for is in [Python/dtoa.c](https://github.com/python/cpython/blob/91a8e002c21a5388c5152c5a4871b9a2d85f0fc1/Python/dtoa.c#L2315).

@mdickinson - Is the approximation used in dtoa applicable to the problem at hand?  If so, do you think that code can be factored out and called here?

Reply:
@abalkin Looks like the relevant code is no longer part of the PR. But the `Python/dtoa.c` code is for floats, and assumes IEEE 754 format; it's not clear to me how it could be used here. (In general, I'm reluctant to add floating-point dependence to code that doesn't need it.)